### PR TITLE
Add cost budget estimation to schema checks

### PR DESCRIPTION
**Pull request by JoelSpeed**
This is the first of two parts that I'm looking to handle to improve CEL cost analysis and bring reporting on it.

This lifts the CEL cost estimation for CEL validation rules into the schema checker so that we can get pre-flight cost validation analysis before applying the CRDs to the cluster.

Stage 2, my intention is find the cases where the cost budget is exceeded, and add additional warnings that point out potential reasons why the cost budget is so long. For example, is it because the list is unbounded, or because the validation itself is massive?

/hold for testing against o/api before merge

### CODE REVIEW

**Comment by JoelSpeed** [comment]
/hold cancel

I think this should be ready to go as a first pass of CEL reporting within the schema checker

File: pkg/manifestcomparators/comp_must_not_exceed_cost_budget.go [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+
+				compResults, err := cel.Compile(
+					typeInfo.Schema,
+					typeInfo.DeclType,
+					celconfig.PerCallLimit,
+					environment.MustBaseEnvSet(environment.DefaultCompatibilityVersion()),
+					cel.NewExpressionsEnvLoader(),
+				)
+				if err != nil {
+					fieldErr := field.InternalError(fldPath, fmt.Errorf("failed to compile x-kubernetes-validations rules: %w", err))
```
Comment (deads2k) [inline pkg/manifestcomparators/comp_must_not_exceed_cost_budget.go:76]: oh this is neat. We get a compilation check now.

Comment (deads2k) [inline pkg/manifestcomparators/comp_must_not_exceed_cost_budget.go:76]: no unit test for this branch?

**Comment by deads2k** [comment]
suggested unit test when you rebase.

/lgtm
/approve

**Pull request merged**